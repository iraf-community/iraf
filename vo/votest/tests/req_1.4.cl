#
#  Demonstrate support for latest VOTable standard spec.

# Set the test description string.
votest.descr = "Demonstrate support for latest VOTable standard spec."

# Convert the data$logical to a local path.
s1 = data_url // "/votable_1.2.xml"
s2 = "file://" // data_path // "/votable_1.2.xml"
s3 = "file:///localhost" // data_path // "/votable_1.2.xml"

print ("------------------------------------------------------------------")
print ("Req  1.4:  VOTable Interface code will support the IVOA Standard")
print ("           specification of the VOTable format at time of release.")
print ("------------------------------------------------------------------")


fcache init

print (" ")
print ("    The requirement is demonstrated using a VOTable v1.2 document")
print ("retrieved from the CADC and modified to add v1.2-specific features:")
print (" ")
print ("    - INFO elements before </TABLE>, </RESOURCE> and </VOTABLE>")
print ("    - INFO elements w/ 'utype', 'ucd', 'ref' and 'unit' attributes")
print ("    - FIELDref and PARAMref elements w/ 'ucd' and 'utype' attributes")
print (" ")
print ("The parse will print an error if these v1.2-specifics aren't supported")
print (" ")
print (" ")


# Execute the test commands.
print ("\nLogical Path:  data$votable_1.2.xml")		# logical path
tlcol ("data$votable_1.2.xml")

print ("\nHTTP URI:  " // s1)				# remote http URI
tlcol (s1)

print ("\nFile URI:  " // s2)				# file URI
tlcol (s2)

print ("\nFile URI:  " // s3)				# file URI
tlcol (s3)

