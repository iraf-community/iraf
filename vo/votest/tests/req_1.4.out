------------------------------------------------------------------
Req  1.4:  VOTable Interface code will support the IVOA Standard
           specification of the VOTable format at time of release.
------------------------------------------------------------------
 
    The requirement is demonstrated using a VOTable v1.2 document
retrieved from the CADC and modified to add v1.2-specific features:
 
    - INFO elements before </TABLE>, </RESOURCE> and </VOTABLE>
    - INFO elements w/ 'utype', 'ucd', 'ref' and 'unit' attributes
    - FIELDref and PARAMref elements w/ 'ucd' and 'utype' attributes
 
The parse will print an error if these v1.2-specifics aren't supported
 
 

Logical Path:  data$votable_1.2.xml
# data$votable_1.2.xml
collection       CH*4         %-4s ""              
collectionID     CH*128     %-128s ""              
instrument_name  CH*128     %-128s ""              
position_center_ra D         %25.16g deg             
position_center_dec D         %25.16g deg             
position_naxes   I            %11d ""              
position_naxis   I            %11d ""              
position_scale   D         %25.16g ""              
position_bounds  CH*193     %-193s deg             
energy_bounds_center D         %25.16g m               
energy_bounds_cval1 D         %25.16g m               
energy_bounds_cval2 D         %25.16g m               
energy_units     CH*32       %-32s ""              
energy_bandpassName CH*32       %-32s ""              
time_bounds_center D         %25.16g d               
time_bounds_cval1 D         %25.16g d               
time_bounds_cval2 D         %25.16g d               
time_exposure    D         %25.16g sec             
imageFormat      CH*128     %-128s ""              
metaRelease      CH*23       %-23s ""              
dataRelease      CH*23       %-23s ""              
accessURL        CH*95       %-95s ""              

HTTP URI:  http://iraf.noao.edu/votest//votable_1.2.xml
# http://iraf.noao.edu/votest//votable_1.2.xml
collection       CH*4         %-4s ""              
collectionID     CH*128     %-128s ""              
instrument_name  CH*128     %-128s ""              
position_center_ra D         %25.16g deg             
position_center_dec D         %25.16g deg             
position_naxes   I            %11d ""              
position_naxis   I            %11d ""              
position_scale   D         %25.16g ""              
position_bounds  CH*193     %-193s deg             
energy_bounds_center D         %25.16g m               
energy_bounds_cval1 D         %25.16g m               
energy_bounds_cval2 D         %25.16g m               
energy_units     CH*32       %-32s ""              
energy_bandpassName CH*32       %-32s ""              
time_bounds_center D         %25.16g d               
time_bounds_cval1 D         %25.16g d               
time_bounds_cval2 D         %25.16g d               
time_exposure    D         %25.16g sec             
imageFormat      CH*128     %-128s ""              
metaRelease      CH*23       %-23s ""              
dataRelease      CH*23       %-23s ""              
accessURL        CH*95       %-95s ""              

File URI:  file:///iraf/iraf/vo/votest/data//votable_1.2.xml
# file:///iraf/iraf/vo/votest/data//votable_1.2.xml
collection       CH*4         %-4s ""              
collectionID     CH*128     %-128s ""              
instrument_name  CH*128     %-128s ""              
position_center_ra D         %25.16g deg             
position_center_dec D         %25.16g deg             
position_naxes   I            %11d ""              
position_naxis   I            %11d ""              
position_scale   D         %25.16g ""              
position_bounds  CH*193     %-193s deg             
energy_bounds_center D         %25.16g m               
energy_bounds_cval1 D         %25.16g m               
energy_bounds_cval2 D         %25.16g m               
energy_units     CH*32       %-32s ""              
energy_bandpassName CH*32       %-32s ""              
time_bounds_center D         %25.16g d               
time_bounds_cval1 D         %25.16g d               
time_bounds_cval2 D         %25.16g d               
time_exposure    D         %25.16g sec             
imageFormat      CH*128     %-128s ""              
metaRelease      CH*23       %-23s ""              
dataRelease      CH*23       %-23s ""              
accessURL        CH*95       %-95s ""              

File URI:  file:///localhost/iraf/iraf/vo/votest/data//votable_1.2.xml
# file:///localhost/iraf/iraf/vo/votest/data//votable_1.2.xml
collection       CH*4         %-4s ""              
collectionID     CH*128     %-128s ""              
instrument_name  CH*128     %-128s ""              
position_center_ra D         %25.16g deg             
position_center_dec D         %25.16g deg             
position_naxes   I            %11d ""              
position_naxis   I            %11d ""              
position_scale   D         %25.16g ""              
position_bounds  CH*193     %-193s deg             
energy_bounds_center D         %25.16g m               
energy_bounds_cval1 D         %25.16g m               
energy_bounds_cval2 D         %25.16g m               
energy_units     CH*32       %-32s ""              
energy_bandpassName CH*32       %-32s ""              
time_bounds_center D         %25.16g d               
time_bounds_cval1 D         %25.16g d               
time_bounds_cval2 D         %25.16g d               
time_exposure    D         %25.16g sec             
imageFormat      CH*128     %-128s ""              
metaRelease      CH*23       %-23s ""              
dataRelease      CH*23       %-23s ""              
accessURL        CH*95       %-95s ""              
